src/freedv_api.c

```diff
@@ -1250,7 +1250,8 @@ void freedv_set_phase_est_bandwidth_mode(struct freedv *f, int val) {
 }
 
 void freedv_set_eq(struct freedv *f, int val) {
-    if (FDV_MODE_ACTIVE( FREEDV_MODE_700C, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700D, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700E, f->mode)) {
+    if (FDV_MODE_ACTIVE( FREEDV_MODE_700C, f->mode) || FDV_MODE_ACTIVE( FREEDV_MODE_700D, f->mode) || 
```

Comment:
Maybe checks like these can be consolidated in the individual Codec2 modes? For instance, `if (codec2_is_mode_700c(f->mode)) { ... }`.

Reply:
:+1:  Yes I was thinking along similar lines, e.g. if we add a field to note the codec 2 mode to the freedv states:
```
if (f->codec2_mode == CODEC2_MODE_700C)
```